### PR TITLE
GPII-1230, GPII-1235, GPII-442: Running application detection and support for more advanced MM

### PR DESCRIPTION
**Pull request by kaspermarkus**
This branch is compatible with the advanced MM branch GPII-1230 of universal (https://github.com/GPII/universal/pull/507). It contains the (relevant) code from Josephs GPII-442 branch

### CODE REVIEW

**Comment by amb26** [comment]
Could we get this updated to master and the conflicts removed? @JavierJF will be looking at this together with his work improving the structure of acceptance tests

**Comment by kaspermarkus** [comment]
Updated things on universal side, so:

ok to test

**Comment by kaspermarkus** [comment]
ok to test

**Comment by kaspermarkus** [comment]
@amb26 With the help of Joseph's fix for GPII-2777 (which is actually a fix for GPII-2668 https://github.com/GPII/windows/pull/156 ) and a bit of other changes, this is now finally passing

File: gpii/node_modules/registeredAT/registeredAT.js [hunk 1]
```diff
@@ -127,6 +127,41 @@ gpii.windows.enableRegisteredAT = function (name, enable, options) {
     return promise;
 };
 
+
+gpii.windows.enableRegisteredAT.queryProcessReporter = function (entry) {
+    return gpii.processReporter.find(entry.options.queryProcess);
```
Comment (klown) [inline gpii/node_modules/registeredAT/registeredAT.js]: This should be routed through `gpii.processReporter.handleIsRunning()`, where the reference to `gpii.processReporter.find()` is given via the `entry` argument.  That is, `entry.options.queryProcess` should look like:

`...
 "queryProcess": [
            {
                "type": "gpii.processReporter.find",
                "command": "Magnify.exe"
            }
  ]
...
`
The  technique is to evaluate a list of criteria to determine is a solution is running, and  to declare how to do that in a solution's entry.

Comment (klown) [reply gpii/node_modules/registeredAT/registeredAT.js]: Further to my comment above, see the "[Special Cases](https://github.com/GPII/universal/tree/master/gpii/node_modules/processReporter#special-cases)" in the ProcessReporter README, specifically the third paragraph

**Comment by klown** [comment]
Hey @kaspermarkus 
I decided to try out [what I was talking about above](https://github.com/GPII/windows/pull/120#discussion_r163582182).  It works, but there were a number of finicky details.  Since I've done it, I decided to make your life easier by branching off your code here, and then making a pull request that you can use.

I'm checking a few things right now, but expect to see something in the next little while, including some associated changes to the universal code (i.e. two pull requests against your code).

**Comment by amb26** [comment]
Hi there @kaspermarkus - could you update this to current master of universal etc. and check that it is still ok?

File: gpii/node_modules/registeredAT/registeredAT.js (hunk 1 above)
Comment (amb26) [reply gpii/node_modules/registeredAT/registeredAT.js]: @kaspermarkus - bump

**Comment by amb26** [comment]
Closed for #173

**Pull request closed without merging**